segmatch/include/segmatch/common.hpp

```diff
@@ -131,10 +132,46 @@ struct IdIndex {
   size_t index = 0u;
 };
 
+static std::string getFileExtension(const std::string& filename) {
+  auto str_iter = filename.rbegin();
+  for (; str_iter != filename.rend(); ++str_iter) {
+    if (*str_iter == '.') break;
+  }
+  return std::string(str_iter.base(), filename.rbegin().base());
```

Comment:
I knew that my function doesn't handle those two cases but didn't care because it is only used in one place and there any string that isn't one of the two file extensions throws an error anyways. 

You're right though, I shouldn't usually do that stuff myself. Especially, I shouldn't put it in a common library potentially for other people to use. 

I didn't know those two std::string functions, thanks!
Since we will move this code to a separate package which will not depend on boost and where the code is in an isolated executable `filename.substr(filename.rfind('.') + 1)` will have to do :)